pkg/operator/controller/externaldns/namespace.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package externaldnscontroller
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// ensureExternalDNSNamespace ensures that the externaldns namespace exists.
+func (r *reconciler) ensureExternalDNSNamespace(namespace string) (bool, *corev1.Namespace, error) {
+	desired := desiredExternalDNSNamespace(namespace)
+
+	haveNamespace, current, err := r.currentExternalDNSNamespace(namespace)
+	if err != nil {
+		return false, nil, err
+	}
+
+	switch {
+	case !haveNamespace:
+		if err := r.createExternalDNSNamespace(desired); err != nil {
+			return false, nil, err
+		}
+		return r.currentExternalDNSNamespace(namespace)
+	}
+
+	return true, current, nil
+}
+
+// currentExternalDNSNamespace  gets the current externalDNS namespace resource.
+func (r *reconciler) currentExternalDNSNamespace(namespace string) (bool, *corev1.Namespace, error) {
+	ns := &corev1.Namespace{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: namespace}, ns); err != nil {
```

Comment:
`context` object can be passed to `ensureExternalDNSNamespace()` (and deeper) from `Reconcile`

Reply:
done! Good idea, thanks for pointing that out.